src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -959,6 +964,21 @@ CreateDistributedExecution(RowModifyLevel modLevel, List *taskList, bool hasRetu
 		ExtractLocalAndRemoteTasks(readOnlyPlan, taskList, &execution->localTaskList,
 								   &execution->remoteTaskList);
 	}
+	else if (jobQuery && list_length(execution->tasksToExecute) == 1)
+	{
+		Task *task = (Task *) linitial(execution->tasksToExecute);
+
+		if (TaskAccessesLocalNode(task))
+		{
```

Comment:
Do we need this condition given that GenerateShardQueryStringIfMissing has an inherent condition? Also, can't we have multiple local tasks?